### PR TITLE
PWN-6593 - fetch prices for new user tokens

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6593

## Description of Work

fetch prices for new user tokens

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/user/repository/UserRemoteRepository.kt [hunk 1]
```diff
@@ -59,17 +60,32 @@ class UserRemoteRepository(
...
             // Load and save user tokens prices
             if (fetchPrices) {
-                val prices = tokenPricesRepository.getTokenPricesBySymbols(
-                    tokenSymbols.map { TokenSymbol(it) },
-                    USD_READABLE_SYMBOL
-                )
-                userLocalRepository.setTokenPrices(prices)
+                loadAndSaveUserTokens(tokenSymbols)
+            } else {
+                checkIsHaveNewToken(tokenSymbols)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/user/repository/UserRemoteRepository.kt:65]: checkForNewTokens

File: app/src/main/java/org/p2p/wallet/user/repository/UserRemoteRepository.kt [hunk 2]
```diff
@@ -59,17 +60,32 @@ class UserRemoteRepository(
...
             }
 
             // Map accounts to List<Token.Active>
             mapAccountsToTokens(publicKey, accounts)
         }
 
+    private suspend fun checkIsHaveNewToken(tokenSymbols: List<String>) {
+        val localPrices = userLocalRepository.getTokenPrices()
+            .firstOrNull()?.map { it.tokenSymbol } ?: listOf()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/user/repository/UserRemoteRepository.kt:74]:
```suggestion
        val localPrices = userLocalRepository.getTokenPrices()
            .firstOrNull()
            ?.map { it.tokenSymbol } 
            .orEmpty()
```

File: app/src/main/java/org/p2p/wallet/user/repository/UserLocalRepository.kt [hunk 3]
```diff
@@ -7,6 +7,7 @@ import org.p2p.wallet.receive.list.TokenListData
 
 interface UserLocalRepository {
     fun setTokenPrices(prices: List<TokenPrice>)
+    fun getTokenPrices(): Flow<List<TokenPrice>>
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/user/repository/UserLocalRepository.kt:10]: isFlowNeeded here? Maybe return just List<>?

Comment (NikolaiEmelianov) [reply app/src/main/java/org/p2p/wallet/user/repository/UserLocalRepository.kt:10]: I think all layers above presenter better to return Flow or use suspend. Flow for reactive, suspend for async.
https://developer.android.com/kotlin/coroutines/coroutines-best-practices#coroutines-data-layer

**Review by kabramovp2p — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**